src/layman/layer/geoserver/wms.py

```diff
@@ -199,16 +200,26 @@ def get_layer_info(workspace, layername, *, x_forwarded_items=None):
 
 def get_layer_info_by_uuid(*, uuid, gdal_layername, gdal_workspace, x_forwarded_items=None):
     gs_layername = names.get_layer_names_by_source(uuid=uuid, ).wms
-    wms = get_wms_proxy()
-    if wms is None or uuid is None:
+    if uuid is None:
         return {}
+    time_info = None
+    wms_layer = gs_util.get_layer(gs_layername.workspace, gs_layername.name, auth=settings.LAYMAN_GS_AUTH)
+    if not wms_layer:
+        return {}
+    if wms_layer['resource']['@class'] == 'coverage' \
+            and get_image_mosaic_store_name(uuid=uuid) in wms_layer['resource']['href']:
+        granules_json = gs_util.get_image_mosaic_granules(gs_layername.workspace,
```

Comment:
I find it a little bit inconsistent, that method returns `granules` but variable is called `granules_json`.